src/Ship/Ship.php

```diff
@@ -0,0 +1,265 @@
+        $this->output->write("");
+    }
+
+    public function chooseShip(Ship $playerShip, Ship $choice): void
+    {
+        $playerShip->setName($choice->getName());
+        $playerShip->setFuel($choice->getFuel());
+        $playerShip->setMaxFuel($choice->getMaxFuel());
+        $playerShip->setHullIntegrity($choice->getHullIntegrity());
+        $playerShip->setMaxHullIntegrity($choice->getMaxHullIntegrity());
+        $playerShip->setShields($choice->getShields());
+        $playerShip->setMaxShields($choice->getMaxShields());
+        $playerShip->setMissileDamage($choice->getMissileDamage());
+        $playerShip->setLaserDamage($choice->getLaserDamage());
+    }
+
+    public function TakeAction(Collection $enemies)
+    {
+        $limit = $enemies->count();
+
+        for ($i = 0; $i < $enemies->count(); $i++) {
+            if ($enemies->get($i)->getHullIntegrity() > 0) {
+                $this->output->write("");
+                $this->output->info("Enemy #" . $i + 1 . " " . $enemies->get($i)->getName());
+                $this->output->info("Shields:" . $enemies->get($i)->getShields());
+                $this->output->info("Hull Integrity:" . $enemies->get($i)->getHullIntegrity());
+            }
+
+        }
+
+        while(true){
+            $target = $this->output->input("Which enemy do you want to target? (Please type the number)", $limit);
+            if($enemies->get($target-1)->getHullIntegrity() > 0){
+                break;
+            }
+        }
+
+        $options = ["Laser" => "Attack with lasers! (Deals " . $this->getLaserDamage() . " damage)", "Missile" => "Attack with missiles! (Deals " . $this->getMissileDamage() . " damage)"];
+        $result = $this->output->getCli()->radio("How do you want to attack him?", $options)->prompt();
+
+        if ($result === "Laser") {
+            $this->playerAttacksWithLaser($enemies->get($target - 1));
+
+        }
+        if ($result === "Missile") {
+            $this->playerAttacksWithMissile($enemies->get($target - 1));
+        }
+    }
+
+    private function takeDamage(int $damage, bool $isLaser)
+    {
+        if ($this->getShields() == 0) {
+            $this->setHullIntegrity(($this->getHullIntegrity() - $damage));
+        }
+
+        if ($this->getShields() > 0) {
+            if (!$isLaser) {
+                $damage = $damage / 2;
+            }
+
+            if ($damage > $this->getShields()) {
+                $damageToHull = $damage - $this->getShields();
+                $this->setHullIntegrity(($this->getHullIntegrity() - $damageToHull));
+            }
+
+            $this->setShields(($this->getShields() - $damage));
+        }
+
+        if ($this->getShields() < 0) {
+            $this->setShields(0);
+        }
+    }
+
+
+    private function playerAttacksWithMissile(Ship $enemyShip)
+    {
+        $enemyShip->takeDamage($this->getMissileDamage(), false);
+    }
+
+    private function playerAttacksWithLaser(Ship $enemyShip)
```

Comment:
Wygląda na to, jakbyś próbował w jednej klasie zmieścić całą logikę związaną z walką. Może lepiej byłoby walkę wyciągnąć poza tę klasę, a jej zostawić jedynie zarządzanie danymi?

Reply:
Wychodziłem z założenia, że skoro to statek strzela w inny statek to mechanika strzelania powinna zostać w klasie statku.